go/pkg/filemetadata/filemetadata.go

```diff
@@ -112,10 +114,12 @@ func Compute(filename string) *Metadata {
 		}
 		xattrValue, err := XattrAccess.getXAttr(filename, XattrDigestName)
 		if err == nil {
-			md.Digest = digest.Digest{
-				Hash: string(xattrValue),
-				Size: file.Size(),
+			xattrStr := string(xattrValue)
+			if strings.Contains(xattrStr, "/") {
+				md.Digest, md.Err = digest.NewFromString(xattrStr)
+				return md
 			}
+			md.Digest, md.Err = digest.NewFromString(fmt.Sprintf("%s/%d", xattrStr, file.Size()))
 			return md
```

Comment:
The previous code with `digest.Digest{}` will create digest with a invalid hash, i.e. "abc/100" (hash length should be 64);
I changed to `digest.NewFromString()` so that if user pass a invalid hash string via xatrr (say, "abc"), md.Err can capture this error.

Reply:
Good catch